### ISSUE TITLE
RetryInterceptor.onError - exceptions handling

### ISSUE THREAD

**Issue by solikhver** [rodion-m/dio_smart_retry#2]
Hi @rodion-m 
I have tried to use your library in my project. I have several tests of my communication layer including test for incorrect url. 
Unfortunately test stops working after adding your interceptor.
Here is what I have discovered.
`RetryInterceptor.onError` do not handling errors that could arise in this method. For example in my case I have had incorrect url (`http    ://github.com` with spaces) that makes troubles when I am specifying `logPrint` function. Interceptor prepare output string for `logPrint` function and call `err.requestOptions.uri` which throws exception `FormatException: Illegal scheme character (at character 5)`. This exception bubble out without corresponding handling. 

The solution could be simple in my case, just use path instead of uri or something similar in that direction. But I propose you to think about general error handling to avoid issues for example when `_retryEvaluator` could throw an error.

**Comment by rodion-m** [comment]
Hi @solikhver ! Thanks a lot for contributing. I accept that we should change `uri` to `path` in logging. \
Also, I think that the default retry evaluator should not retry this type of error at all. So, I plan to add something like this in a default retry evaluator:
```dart
...
      shouldRetry = error.type != DioErrorType.cancel
                  && error.error is! FormatException;
...
```

**Comment by solikhver** [comment]
As for me you can wrap any exception that could happens during `RetryInterceptor.onError` call with your own and do not retry if this type of error is presented. But definitely it shouldn't be unhandled exceptions even if it occurs in evaluator.

**Comment by rodion-m** [comment]
The issue is fixed #[2af5c2b](https://github.com/rodion-m/dio_smart_retry/commit/2af5c2b45759ddb939da0dab4631aec96b6d0c87). \
Also, a new parameter  `ignoreRetryEvaluatorExceptions` is added by default it's `false`. \
Thanks again for contributing!
